### DISCUSSION TITLE
branching plugin in netbox/docker envirnonemnt

### DISCUSSION THREAD

**Discussion by CooperCOK** [netboxlabs/netbox-branching#161] [Q&A] [answered]
Is there a way to install and activate the branching plugin in a netbox/docker environment? if so, how? thx

**Comment by sebastianreifetshammer** [comment, marked as answer]
using [netbox-docker](https://github.com/netbox-community/netbox-docker) github repo:

I've set up a docker-compose.override.yml file, which is tracked in my private GitLab instance and is symlinked to the netbox-docker directory where the compose project is started. Similarly, my custom Dockerfile-Plugins is also symlinked to this directory. Along with this, plugins.py and local_settings.py are bind-mounted into the container through the override file.

A key detail to keep in mind: make sure local_settings.py is outside of the configuration directory, as the file gets overridden by the bind mount in the default docker-compose.yml. To address this, you can add the line COPY local_settings.py /opt/netbox/netbox/netbox/local_settings.py to your Dockerfile, ensuring it's placed in the correct location within the container. Also make sure the package is installed in virtual environment in the container, I made a requirements file and added it to my Dockerfile (RUN /opt/netbox/venv/bin/pip install  --no-warn-script-location -r /opt/netbox/plugin_requirements.txt)

For more details, refer to [this GitHub discussion](https://github.com/netbox-community/netbox-docker/discussions/1316) on the NetBox Docker setup.

Overall, the setup closely follows the installation guide. I successfully activated the plugin this way, though I ran into an issue with job execution on a new branch, specifically an error message stating, "The last job (Provision branch) failed with an error." However, it might work smoothly on your side.
